core/arch/arm/plat-ls/main.c

```diff
@@ -94,3 +167,36 @@ void console_flush(void)
 {
 	ns16550_flush(console_base());
 }
+
+void main_init_gic(void)
+{
+	vaddr_t gicc_base;
+	vaddr_t gicd_base;
+
+	gicc_base = (vaddr_t)phys_to_virt(GIC_BASE + GICC_OFFSET,
+					  MEM_AREA_IO_SEC);
+	gicd_base = (vaddr_t)phys_to_virt(GIC_BASE + GICD_OFFSET,
+					  MEM_AREA_IO_SEC);
+
+	if (!gicc_base || !gicd_base)
+		panic();
+
+	/* Initialize GIC */
+	gic_init(&gic_data, gicc_base, gicd_base);
+
+	itr_init(&gic_data.chip);
+}
+
+static void platform_tee_entry_fast(struct thread_smc_args *args)
```

Comment:
I am not sure, this function happens to be identical for both i.MX6 and LS102x; but on ZC702, it extends this function to handle additional platform defined SMCs. So this function is intended to be extended or overwritten for each platform, it may call other "generic" functions (e.g., `the generic_smp_boot`) if that platform happens to support a common SMC request, but the function overall is platform dependent thus not generic.